src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs

```diff
@@ -43,16 +44,38 @@ public HttpsConnectionMiddleware(ConnectionDelegate next, HttpsConnectionAdapter
                 throw new ArgumentNullException(nameof(options));
             }
 
+            _options = options;
+            _logger = loggerFactory.CreateLogger<HttpsConnectionMiddleware>();
+
             // This configuration will always fail per-request, preemptively fail it here. See HttpConnection.SelectProtocol().
             if (options.HttpProtocols == HttpProtocols.Http2)
             {
                 if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
                 {
                     throw new NotSupportedException(CoreStrings.HTTP2NoTlsOsx);
                 }
-                else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && Environment.OSVersion.Version < new Version(6, 2))
```

Comment:
Is there any risk that someone got ALPN working on Windows 8 somehow and this change breaks them despite the AppContext switch?

Reply:
This probably warrants a breaking changes announcement.

Reply:
This definitely warrants a breaking change announcement.

If someone does have ALPN working on Windows 8, changing `< new Version(6, 2)` to `< new Version(6, 3)` would completely break them and not even the AppContext switch would work around the issue.

Reply:
ALPN support was added in Windows 8.1 so I don't think that was ever a scenario that worked. I made the change since I noticed that Windows 8.1 which was where ALPN support was first added is version 6.3 instead of 6.2. Unless there was an actual reason for allowing Http2 on Windows 8?

Reply:
Ping @Tratcher

Reply:
No reason to allow it on Win8 without ALPN, it's the same as Win7.